### PR TITLE
[ci] Run analysis with older versions of Flutter

### PR DESCRIPTION
**Pull request by stuartmorgan**
As a sanity check to reduce accidentally publishing plugins that claim to work on versions we no longer officially support, but with Flutter pubspec constraints that will cause them to be picked up by old versions, this runs `analyze` with the previous two stable releases. While this won't catch all possible issues (runtime failures, use of new Flutter APIs in native code), it will catch a whole category of such errors with relatively little additional CI cost.

While this does require manual updates to pubspec.yaml periodically, since this is best-effort testing such updates aren't time-critical, as the impact of forgetting is minimal.

Restructures the tool's analyze command to run `flutter packages get` during package iteration, instead of pre-iterating everything, to allow for skipping packages by version without having to duplicate logic. (This also has the advantage that failures in that step will be assigned to the specific package that causes the problem, as with most failures, rather than a generic step as it was previously doing.)

No version change: Dropping support for older versions of Flutter doesn't need a release, just to be applied to future releases.

Part of https://github.com/flutter/flutter/issues/98697

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [relevant style guides] and ran [the auto-formatter]. (Unlike the flutter/flutter repo, the flutter/plugins repo does use `dart format`.)
- [x] I signed the [CLA].
- [x] The title of the PR starts with the name of the plugin surrounded by square brackets, e.g. `[shared_preferences]`
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I updated `pubspec.yaml` with an appropriate new version according to the [pub versioning philosophy], or this PR is [exempt from version changes].
- [x] I updated `CHANGELOG.md` to add a description of the change, [following repository CHANGELOG style].
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] All existing and new tests are passing.

[Contributor Guide]: https://github.com/flutter/plugins/blob/master/CONTRIBUTING.md
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[relevant style guides]: https://github.com/flutter/plugins/blob/master/CONTRIBUTING.md#style
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat
[pub versioning philosophy]: https://dart.dev/tools/pub/versioning
[exempt from version changes]: https://github.com/flutter/flutter/wiki/Contributing-to-Plugins-and-Packages#version-and-changelog-updates
[following repository CHANGELOG style]: https://github.com/flutter/flutter/wiki/Contributing-to-Plugins-and-Packages#changelog-style
[the auto-formatter]: https://github.com/flutter/plugins/blob/master/script/tool/README.md#format-code
[test-exempt]: https://github.com/flutter/flutter/wiki/Tree-hygiene#tests

### CODE REVIEW

**Comment by stuartmorgan** [comment]
This should be ready for review (I'll just need to re-push to get `publishable` to update). There are a couple of cases where we have to drop support for older versions only because of the example or integration test using new functionality, which isn't ideal, but if we can't run analyze on those versions then we can't get the safety check this is meant to provide, so dropping support seems like the best option.

**Review by bparrishMines — APPROVED** [review]
LGTM with a nit

File: .cirrus.yml [hunk 1]
```diff
@@ -138,6 +138,34 @@ task:
...
+    # Note: The versions below should be manually updated after a new stable
+    # version comes out.
+    - name: legacy-version-analyze
+      depends_on: analyze
+      env:
+        matrix:
+          CHANNEL: "2.5.3"
+          CHANNEL: "2.8.1"
+      analyze_script:
+        - ./script/tool_runner.sh analyze --skip-not-supporting-flutter-version="$CHANNEL" --custom-analysis=script/configs/custom_analysis.yaml
```
Comment (ditman) [inline .cirrus.yml:155]: This is great stuff! Thanks for adding this configurability to the package_looping_command!

**Review by ditman — APPROVED** [review]

File: script/tool/lib/src/common/package_looping_command.dart [hunk 2]
```diff
@@ -75,7 +77,16 @@ abstract class PackageLoopingCommand extends PluginCommand {
...
+            processRunner: processRunner, platform: platform, gitDir: gitDir) {
+    argParser.addOption(
+      _skipByFlutterVersionArg,
+      help: 'Skip any packages that require a Flutter version newer than '
+          'the provided version.',
+    );
+  }
+
+  static const String _skipByFlutterVersionArg =
+      'skip-not-supporting-flutter-version';
```
[the remark this answers is not in the document]
Comment (stuartmorgan) [reply script/tool/lib/src/common/package_looping_command.dart:89]: Yes, that reads a little better; done.

**Pull request merged**